### PR TITLE
SAK-50492 Meetings allow instructors to be co-organizers in online meeting

### PR DESCRIPTION
**Pull request by Aniii10**
We want to allow that when an instructor or organizer creates a meeting, not only can he start the meeting, but it can also be started by another instructor who will assume the role of co-organizer of the meeting.

### CODE REVIEW

File: meetings/tool/src/main/java/org/sakaiproject/meetings/controller/MeetingsController.java [hunk 1]
```diff
@@ -18,17 +18,13 @@
...
-import java.util.Collection;
-import java.util.Collections;
-import java.util.List;
-import java.util.Locale;
-import java.util.Optional;
-import java.util.Set;
+import java.util.*;
 import java.util.stream.Collectors;
 
+import lombok.Setter;
```
Comment (bgarciaentornos) [inline meetings/tool/src/main/java/org/sakaiproject/meetings/controller/MeetingsController.java:24]: It seems this is not used

File: meetings/tool/src/main/java/org/sakaiproject/meetings/controller/MeetingsController.java [hunk 2]
```diff
@@ -18,17 +18,13 @@
...
 import java.time.Instant;
 import java.time.format.DateTimeParseException;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.List;
-import java.util.Locale;
-import java.util.Optional;
-import java.util.Set;
+import java.util.*;
```
Comment (bgarciaentornos) [inline meetings/tool/src/main/java/org/sakaiproject/meetings/controller/MeetingsController.java:21]: Please keep just the used imports

File: meetings/tool/src/main/java/org/sakaiproject/meetings/controller/MeetingsController.java [hunk 3]
```diff
@@ -390,8 +389,26 @@ public Meeting createMeeting(@RequestBody MeetingData data) throws MeetingsExcep
...
 			String onlineMeetingUrl = null;
+			List<String> coorganizerEmails = new ArrayList<>();
 			if (MS_TEAMS.equals(data.getProvider())) {
-				TeamsMeetingData meetingTeams = microsoftCommonService.createOnlineMeeting(user.getEmail(), meeting.getTitle(), meeting.getStartDate(), meeting.getEndDate());
+				if (data.isCoorganizersEnabled()) {
+					List<Member> coorganizers = sakaiProxy.getSite(data.getSiteId()).getMembers()
+							.stream()
+							.filter(u -> {
+								boolean canUpdate = sakaiProxy.canUpdateSite("/site/" + data.getSiteId(), u.getUserId());
+								log.info("User: " + u.getUserId() + " canUpdate: " + canUpdate);
```
Comment (bgarciaentornos) [inline meetings/tool/src/main/java/org/sakaiproject/meetings/controller/MeetingsController.java:399]: debug

File: meetings/tool/src/main/java/org/sakaiproject/meetings/controller/MeetingsController.java [hunk 4]
```diff
@@ -390,8 +389,26 @@ public Meeting createMeeting(@RequestBody MeetingData data) throws MeetingsExcep
...
+								return canUpdate;
+							})
+							.collect(Collectors.toList());
+
+					coorganizers.forEach(c -> log.info("Filtered Coorganizer: " + c.getUserId()));
```
Comment (bgarciaentornos) [inline meetings/tool/src/main/java/org/sakaiproject/meetings/controller/MeetingsController.java:404]: debug

File: meetings/tool/src/main/java/org/sakaiproject/meetings/controller/MeetingsController.java [hunk 5]
```diff
@@ -390,8 +389,26 @@ public Meeting createMeeting(@RequestBody MeetingData data) throws MeetingsExcep
...
+
+					coorganizerEmails = coorganizers.stream()
+							.map(member -> sakaiProxy.getUser(member.getUserId()).getEmail())
+							.filter(email -> email != null && !email.isEmpty())
```
Comment (bgarciaentornos) [inline meetings/tool/src/main/java/org/sakaiproject/meetings/controller/MeetingsController.java:408]: You can use StringUtils.isNotEmpty which checks both cases

File: meetings/tool/src/main/resources/create-meeting_eu.properties [hunk 6]
```diff
@@ -16,6 +16,7 @@ no_notification= Bat ere ez - Jakinarazpenik gabe
 notifications= Jakinarazpenak
 open_date= Hasteko data
 participants_selection= Partaideen hautaketa
+enable_coorganizers= Gehitu guneko irakasleak ko-antolatzaile gisa
```
Comment (bgarciaentornos) [inline meetings/tool/src/main/resources/create-meeting_eu.properties:19]: @pablosanroman please can you check? Thanks!

**Review by bgarciaentornos — CHANGES_REQUESTED** [review]
Great work, thanks! Just some small things

File: meetings/tool/src/main/resources/create-meeting_eu.properties (hunk 6 above)
Comment (PabloSanRoman) [reply meetings/tool/src/main/resources/create-meeting_eu.properties:19]: hi   @bgarciaentornos! here it is: 
enable_coorganizers= Gehitu guneko irakasleak antolatzaile-kide gisa

Comment (bgarciaentornos) [inline meetings/tool/src/main/resources/create-meeting_eu.properties:19]:
```suggestion
enable_coorganizers= Gehitu guneko irakasleak antolatzaile-kide gisa
```

**Review by bgarciaentornos — APPROVED** [review]
Thanks, this looks ready to go!

**Pull request merged**